### PR TITLE
Migrate and adapt macos wazuh agent packages subsystem

### PR DESCRIPTION
**Pull request by mjcr99**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/22080|

## Description
As the main task of this PR is to move all required components from the wazuh-package repository to be able to generate RPM, DEB, ZIP, and PKG packages for the agent, the following steps were taken:

- Moved the wazuh-agent specs from the wazuh-packages repository to the wazuh repository.
- Improved the generate_wazuh_package.sh script.
- Added workflow for Macos package generation. 
- Used the sources `branch` to specify that the specs and source code will be taken from the downloaded branch.
- 
- Update the package name to adhere to [this naming convention](https://github.com/wazuh/wazuh/issues/21755#:~:text=As%20subsystem%20owners%2C%20teams%20must%20follow%20this%20naming%20convention%20for%20development%20package%20names%3A%20%3CSubsystemName%3E_%3CVersionNumber%3E%2D%3CRevision%3E_%7B%3COSName%3E%2D%3COSVersion%3E_%7D%3CArchitecture%3E_%3CGitReference%3E.%3CPackageType%3E.) for development package names:
```
 <SubsystemName>_<VersionNumber>-<Revision>_{<OSName>-<OSVersion>_}<Architecture>_<GitReference>.<PackageType>.
```

## Generate package
To generate a DEB/RPM package for wazuh-agent:

```
git clone --branch 22080-migrate-and-adapt-macos-wazuh-agent-subsystem-1
https://github.com/wazuh/wazuh.git
cd wazuh/packages/macos

./generate_wazuh_package.sh -s /tmp  -a [arch] -r [revision]  
```

## Tests

- Compilation without warnings in every supported platform
  - [ ] Linux
  - [ ] Windows
  - [x] MAC OS X

### CODE REVIEW

File: .github/workflows/packages-build-macos-agent.yml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        default: "0"
+      is_stage:
+        type: boolean
+        description: |
+          Should set development/production nomenclature
+          True if package name should have production format.
+          False if package name should have developer format.
+          Default is 'false'.
+        required: false
+        default: false
```
Comment (lchico) [inline .github/workflows/packages-build-macos-agent.yml:27]: These lines can be safely removed as they are already set to false by default. Leaving a comment explaining this (as you've done) is sufficient.

File: .github/workflows/packages-build-macos-agent.yml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+          if [ "${{ inputs.checksum }}" == "true" ]; then FLAGS+="--checksum "; fi
+          if [ "${{ inputs.debug}}" == "true" ]; then FLAGS+="--debug "; fi
+          sudo bash generate_wazuh_packages.sh -a ${{ inputs.architecture }} $FLAGS
+
+      - name: Test install Wazuh agent
+        working-directory: packages/macos/output
+        run: |
+          sudo echo "WAZUH_MANAGER='1.1.1.1'" > /tmp/wazuh_envs && sudo installer -pkg *agent*pkg -target /
+          sudo /Library/Ossec/bin/wazuh-control start
+          if [ $(sudo /Library/Ossec/bin/wazuh-control status | grep 'is running...' | wc -l) -gt "0" ];
```
Comment (lchico) [inline .github/workflows/packages-build-macos-agent.yml:89]:
```suggestion
          if [ $(sudo /Library/Ossec/bin/wazuh-control status | grep 'is running...' | wc -l) -gt 5 ];
```
So, we validate that all services are running:

```
wazuh-modulesd is running...
wazuh-logcollector is running...
wazuh-syscheckd is running...
wazuh-agentd is running...
wazuh-execd is running...
```

File: packages/macos/generate_wazuh_packages.sh [hunk 3]
```diff
@@ -123,16 +125,32 @@ function get_pkgproj_specs() {
...
     # Download source code
-    git clone --depth=1 -b ${BRANCH_TAG} ${WAZUH_SOURCE_REPOSITORY} "${WAZUH_PATH}"
-    short_commit_hash="$(curl -s https://api.github.com/repos/wazuh/wazuh/commits/${BRANCH_TAG} \
-                          | grep '"sha"' | head -n 1| cut -d '"' -f 4 | cut -c 1-7)"
+    if [ -n "$BRANCH_TAG" ]; then
+        SOURCES_DIRECTORY="${CURRENT_PATH}/repository"
+        WAZUH_PATH="${SOURCES_DIRECTORY}/wazuh"
+        git clone --depth=1 -b ${BRANCH_TAG} ${WAZUH_SOURCE_REPOSITORY} "${WAZUH_PATH}"
+        short_commit_hash="$(curl -s https://api.github.com/repos/wazuh/wazuh/commits/${WAZUH_BRANCH} \
+                | grep '"sha"' | head -n 1| cut -d '"' -f 4 | cut -c 1-11)"
```
Comment (lchico) [inline packages/macos/generate_wazuh_packages.sh:133]: Could you check if this works:
```suggestion
short_commit_hash=$(git rev-parse -C "${WAZUH_PATH}" --short HEAD)
```

File: packages/macos/package_files/build.sh [hunk 4]
```diff
@@ -11,6 +11,7 @@ set -exf
 DESTINATION_PATH=$1
 SOURCES_PATH=$2
 BUILD_JOBS=$3
+DEBUG_ENABLED=$4
```
Comment (lchico) [inline packages/macos/package_files/build.sh:14]: Could we just use DEBUG? It seems like it's always the same flag, and the current name is a bit confusing.

**Review by lchico — APPROVED** [review]
I left some minor comments in the code. Feel free to ignore them.

**Review by lchico — CHANGES_REQUESTED** [review]
Looks great! To improve validation, let's add checks for the CUSTOM_CODE_VOL path.

**Review by lchico — APPROVED** [review]

File: .github/workflows/packages-build-macos-agent.yml [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        required: false
+      debug:
+        type: boolean
+        required: false
+      checksum:
+        type: boolean
+        required: false
+
+jobs:
+  Build-agent-linux-packages:
```
Comment (MarcelKemp) [inline .github/workflows/packages-build-macos-agent.yml:57]:
```suggestion
  build-agent-macos-packages:
```

File: .github/workflows/packages-build-macos-agent.yml [hunk 6]
```diff
@@ -0,0 +1,105 @@
+name: Build Macos Wazuh Agent Packages - intel64|arm64
```
Comment (MarcelKemp) [inline .github/workflows/packages-build-macos-agent.yml:1]:
```suggestion
name: Build macOS Wazuh Agent Packages - intel64|arm64
```

File: .github/workflows/packages-build-macos-agent.yml [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+        type: boolean
+        required: false
+      checksum:
+        type: boolean
+        required: false
+
+jobs:
+  Build-agent-linux-packages:
+    runs-on: macos-latest
+    name: Build MacOS wazuh-agent on ${{ inputs.architecture }}
```
Comment (MarcelKemp) [inline .github/workflows/packages-build-macos-agent.yml:59]:
```suggestion
    name: Build macOS wazuh-agent on ${{ inputs.architecture }}
```

**Review by MarcelKemp — CHANGES_REQUESTED** [review]
GJ, just a few minor changes.

File: .github/workflows/packages-build-macos-agent.yml (hunk 5 above)
Comment (jotacarma90) [reply .github/workflows/packages-build-macos-agent.yml]: done

File: .github/workflows/packages-build-macos-agent.yml (hunk 6 above)
Comment (jotacarma90) [reply .github/workflows/packages-build-macos-agent.yml]: done

File: .github/workflows/packages-build-macos-agent.yml (hunk 5 above)
Comment (jotacarma90) [reply .github/workflows/packages-build-macos-agent.yml]: done

File: packages/macos/generate_wazuh_packages.sh (hunk 3 above)
Comment (jotacarma90) [reply packages/macos/generate_wazuh_packages.sh]: done

**Review by MarcelKemp — APPROVED** [review]

**Comment by jotacarma90** [comment]
## Builds successfully with different params

🟢 https://github.com/wazuh/wazuh/actions/runs/8569902622
🟢 https://github.com/wazuh/wazuh/actions/runs/8569904263
🟢 https://github.com/wazuh/wazuh/actions/runs/8569907318

**Pull request merged**